Gems/Atom/RHI/Code/Include/Atom/RHI/FrameGraphCompiler.h

```diff
@@ -162,14 +162,21 @@ namespace AZ
 
             void CompileResourceViews(const FrameGraphAttachmentDatabase& attachmentDatabase);
 
-            //Returns the resource from local cache if it exists within it or create one if it doesn't and add it to the cache
+            //! Remove the entry related to the provided attachmentId/idViewHash from the appropriate cache as it is probably stale now
+            template<typename ObjectType>
+            void RemoveFromCache(RHI::AttachmentId attachmentId, HashValue64 idViewHash, ObjectCache<ObjectType>& cache);
+
+            // Returns the resource from local cache if it exists within it or create one if it doesn't and add it to the cache
             ImageView* GetImageViewFromLocalCache(Image* image, const ImageViewDescriptor& imageViewDescriptor);
             BufferView* GetBufferViewFromLocalCache(Buffer* buffer, const BufferViewDescriptor& bufferViewDescriptor);
             
             // This cache is mainly for transient resources. It adds a dependency to the resource views and hence they wont be
             // deleted at the end of the frame and re-created at the start. Mainly used as an optimization.  
             ObjectCache<ImageView> m_imageViewCache;
             ObjectCache<BufferView> m_bufferViewCache;
+            // This map is used to reverse look up view hashes so we can clear them out of m_imageViewCache/m_bufferViewCache
+            // once they have been replaced with a new view instance. 
+            AZStd::unordered_map<HashValue64, HashValue64> m_reverseLookupHash;
```

Comment:
I'm moderately worried about collisions actually. Can we store lookups by name instead which would be hashed to accelerate map search but does an additional equality check after?

Reply:
Another option is to change the hash to a struct containing the name and view descriptor so that we are able to do a equality check in case of a collision

Reply:
I agree it's generally unlikely, but that assumes that hash is "good quality." I'm less worried about incorrectly removing a view, and more keeping a view around that should be invalidated maybe. I guess my suggestion is just, why not key it to `image->GetName()` directly?